src/calculate/pass_4/rah.rs

```diff
@@ -0,0 +1,199 @@
+use crate::calculate::item::Object;
+use crate::calculate::pass_3::Cache;
+use crate::calculate::Ship;
+use crate::info::Info;
+
+const RAH_TYPE_ID: i32 = 4403;
+const RESISTANCE_SHIFT_AMOUNT_ID: i32 = 1849;
+const RESONANCE_IDS: [i32; 4] = [267, 268, 269, 270]; // Order: Em, Explo, Kin, Therm
+
+pub fn adapt_rah_to_damage_pattern(info: &impl Info, ship: &mut Ship) {
```

Comment:
As your work is based on the work of PyFa, please credit them. Something alone the lines of:

`/* Inspires by the Effect4928 class in PyFa. */`

This keeps things fair to them, and clear to us :)